library/std/build.rs

```diff
@@ -3,6 +3,9 @@ use std::env;
 fn main() {
     println!("cargo:rerun-if-changed=build.rs");
     let target = env::var("TARGET").expect("TARGET was not set");
+    if target.contains("netbsd") && env::var("RUST_STD_NETBSD10").is_ok() {
```

Comment:
Since https://github.com/rust-lang/rust/pull/120232, this should look at `target_os`.